### PR TITLE
[requirements] Added Django 3 support, upgraded dependencies

### LINKED ISSUE #79: [requirements] Update dependencies
We need to update all the dependencies including update to django 3.0

### PR DESCRIPTION
**Pull request by pawelplsi**
Dropped  Python 2 and Django<2.2 #79
Closes #79

### CODE REVIEW

**Review by nemesifier — COMMENTED** [review]
@pawelplsi shorten the commit message please, it's too long

File: .travis.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ branches:
     - master
 
 env:
+  - DJANGO="django>=2.2,<3.0"
+  - DJANGO="django>=3.0,<3.3"
```
Comment (nemesifier) [inline .travis.yml]: add a blank line after that line

**Review by nemesifier — COMMENTED** [review]
remove python 3.5 and remove the if `TRAVIS_PYTHON_VERSION == 3.5`

File: .travis.yml (hunk 1 above)
Comment (pawelplsi) [reply .travis.yml]: I think it's not longer relevant, please see how it is now

**Comment by pawelplsi** [comment]
Ofc, I checked the admin subpages (including export/import views) and spotted no error in console :smile:

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
-django>=1.11,<2.3
+django>=2.2,<3.3
```
Comment (atb00ker) [inline requirements.txt]: django 3.3 ? Do you mean to put django 3.1 here? :smile:

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
 swapper>=1.1.0
-openwisp-utils>=0.3.2,<0.4
+openwisp-utils>=0.3.2,<0.5
```
Comment (nemesifier) [inline requirements.txt]: set 0.4 as minimum please

File: requirements.txt (hunk 2 above)
Comment (nemesifier) [reply requirements.txt]: @atb00ker is right, please correct

**Review by nemesifier — CHANGES_REQUESTED** [review]
Almost there, please implement these minor corrections

**Review by atb00ker — COMMENTED** [review]
I also see some `from __future__ import unicode_literals`, we don't need them now! :smile:

File: requirements.txt (hunk 2 above)
Comment (pawelplsi) [reply requirements.txt]: Yeah, my mistake

File: .travis.yml [hunk 4]
```diff
@@ -14,33 +13,24 @@ branches:
...
 
 install:
-  - pip install "django>=2.0,<2.3"
+  - pip install $DJANGO
   - python setup.py -q develop
 
 script:
   - >
     openwisp-utils-qa-checks --migration-path "./django_ipam/migrations/ ./tests/sample_ipam/migrations/"
     --skip-checkmakemigrations
```
Comment (nemesifier) [inline .travis.yml]: you either should remove `--skip-checkmakemigrations` or you should not remove the check below (just remove the if, not the check).

**Review by nemesifier — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 4 above)
Comment (pawelplsi) [reply .travis.yml]: Okay, I'll correct that, but could you tell me why was that check executed previously only for one build, that with python 3.5? I don't get it.

**Comment by pawelplsi** [comment]
@nemesisdesign The if is then actually needed there but only without python version check since the migration tests will fail with DEMO_APP

File: .travis.yml (hunk 4 above)
Comment (atb00ker) [reply .travis.yml]: Because 2.7 on travis had some issue if I recall correctly! :smile:

**Review by nemesifier — APPROVED** [review]

**Comment by nemesifier** [comment]
> @nemesisdesign The if is then actually needed there but only without python version check since the migration tests will fail with DEMO_APP

Yes correct, thanks!

**Pull request merged**